use_cluster.py

```diff
@@ -366,4 +355,411 @@ def return_cluster_dict(q_cluster, q_ids_list):
+            modified_clus_qs.append(n_grams_for_question) #Adding modified question to new formed cluster of modified questions
+            og_clus_qs.append(documents[q_index]._.raw_text)
+        clus_centroid = np.mean(clus_embeddings, axis=0)
+        centroids.append(clus_centroid)
+        modified_clus_list.append(modified_clus_qs)
+    
+        cluster_data["centroid"] = clus_centroid
+        cluster_data["keyword_questions"] = modified_clus_qs
+        cluster_data["original_clustered_qs"] = og_clus_qs
+        cluster_objects[cluster_index] = cluster_data
+    
+    return cluster_objects, lemmatized_qs_list
+    
+"""
+> The objective of this function is to:
+1. Generate list of all unique keywords from corpus
+    
+2. Keywords are used for labelling, and are of form of n-grams
+
+3. Output is a list of n-gram keywords
+    >E.g for 2 grams o/p is a list of all 2-gram keywords of corpus
+    
+: The inputs required are Documents (Question Objects) and Keyword Length to be considered 
+: Keyword Length decides if each keyword is a 1-gram, 2-gram, etc. 
+"""
+
+def generate_global_keywords(documents, keyword_length):
+    num_docs = len(documents)
+    global_keywords = set() #List to store all keywords in corpus (each keyword here is a n-gram, where n is determined by keyword_length)
+    for i in range(num_docs):
+        logging.debug(documents[i])
+        lemma_doc = documents[i]._.lemma_list
+        keywords = generate_ngrams(keyword_length, lemma_doc)
+        global_keywords.update(keywords)
+    global_keywords = set(x.upper() for x in global_keywords)
+    return global_keywords
+
+
+
+"""
+> The objective of this function is to:
+1. Calculate the distances for each keyword from the centroid, to be used for scoring
+    
+2. Call the custom metric function to calculate scores for labels 
+    > Call NMI_Metric
+    > Add distances term with the factor to normalise, to get the custome score
+
+3. Return list of Top 5 Label Scores for each cluster
+
+:The inputs required are taken from the outputs of the generate_modified_qs_data function 
+: The cluster object has attributes: centroid of cluster, raw text questions, lemmatized questions
+"""
+
+def calculate_label_score(cluster_objects, vecs, global_keywords, lemmatized_qs_list):
+    #For each term and each cluster, get frequencies over all qs present in our corpus (NMI)
+    num_clus = len(cluster_objects)
+    all_cluster_scores = []
+    for cluster_index in range(num_clus):
+        modified_clus_qs = cluster_objects[cluster_index]["keyword_questions"] #clus_qs list of qs from the modified cluster 
+        og_clus_qs = cluster_objects[cluster_index]["original_clustered_qs"]
+        centroid = cluster_objects[cluster_index]["centroid"]
+        vec_to_centroid = centroid - vecs
+        distances = np.linalg.norm(vec_to_centroid, axis=1)
+        logging.debug("Total number of Keywords:",len(distances))
+        gloabl_min_distance = min(distances)
```

Comment:
Let's fix the misspelling in the name (making it "global_min_distance"). It's currently misspelled every time, so the code runs fine! But it'll read cleaner fixed.

Reply:
Yes, thank you for catching that! :D